### ISSUE TITLE
Add a helper to change CRAN repos url in package.lock

### ISSUE THREAD

**Issue by cderv** [rstudio/packrat#425]
#378 discussion @kevinushey said about deployement of packrat project and change of CRAN mirror: 
> One way would be to modify the Packrat lockfile at deploy time, to change the repositories to whatever is desired.

It is the way we go when deploying code bundle with packrat. We often need to modify the CRAN repos indicated in `packrat.lock` to change the CRAN-like repos used when restoring. We do that in _bash_ with `sed` or in R directly in custom deployement function (wrappers that include changes in environment variable too.

Would it be be accepted to have a helper fonction in `packrat` to make this step easier when deploying code ? There are helpers for setting cran-like repo (`repo_add`, `repo_set` and friends) but not for changing `packrat.lock` without another snapshot.

Currently, I use something like this : 

```r
change_packrat_repos <- function(cran_url) {
    lockfile <- packrat:::readDcf("packrat/packrat.lock")
    lockfile[1, "Repos"] <- paste("CRAN", cran_url, sep = "=") 
    packrat:::write_dcf(lockfile, file = "packrat/packrat.lock")
}
```

There seems to be other use case for modifying `packrat.lock` repos field like #227 . 

Thanks. Happy to contribute and help.

**Comment by kevinushey** [comment]
I would be open to this. Perhaps:

    packrat::set_lockfile_repos(repos)

or

    packrat::set_lockfile_metadata(repos = repos)

**Comment by cderv** [comment]
Glad your are interested! 👍 
I agree with you on the second API and I will PR a first attempt soon based on your suggestion.
Thanks!

**Comment by cderv** [comment]
closed by #429
